python/ray/tune/ray_trial_executor.py

```diff
@@ -208,32 +237,36 @@ def start_trial(self, trial, checkpoint=None):
             checkpoint (Checkpoint): A Python object or path storing the state
                 of trial.
         """
-
         self._commit_resources(trial.resources)
         try:
             self._start_trial(trial, checkpoint)
-        except Exception as e:
-            logger.exception("Error starting runner for Trial %s", str(trial))
+        except AbortTrialExecution:
+            logger.exception("Trial %s: Error starting runner, aborting!",
+                             trial)
+            time.sleep(2)
             error_msg = traceback.format_exc()
+            self._stop_trial(trial, error=True, error_msg=error_msg)
+            return  # don't retry fatal Tune errors
+        except Exception:
+            logger.exception(
+                "Trial %s: Error starting runner. Attempting "
+                "restart without checkpoint.", trial)
             time.sleep(2)
+            error_msg = traceback.format_exc()
             self._stop_trial(trial, error=True, error_msg=error_msg)
-            if isinstance(e, AbortTrialExecution):
-                return  # don't retry fatal Tune errors
             try:
                 # This forces the trial to not start from checkpoint.
                 trial.clear_checkpoint()
-                logger.info(
-                    "Trying to start runner for Trial %s without checkpoint.",
-                    str(trial))
                 self._start_trial(trial)
             except Exception:
                 logger.exception(
-                    "Error starting runner for Trial %s, aborting!",
-                    str(trial))
+                    "Trial %s: Error starting runner on second "
+                    "attempt, aborting!", trial)
                 error_msg = traceback.format_exc()
                 self._stop_trial(trial, error=True, error_msg=error_msg)
-                # note that we don't return the resources, since they may
-                # have been lost
+                # Note that we don't return the resources, since they may
+                # have been lost.
+                # TODO(ujvl): is this the right thing to do?
```

Comment:
Oh hmmmm this might actually be one of the issues that we're seeing https://github.com/ray-project/ray/issues/5189 and https://github.com/ray-project/ray/issues/5262

Reply:
agree